book/src/getting-started/5-cmake-integration.md

```diff
@@ -26,58 +23,71 @@ edition = "2021"
+
+## CMake setup
+
+Start the CMakeLists.txt file like any other C++ project using Qt. For this example, we are [supporting both
+Qt5 and Qt6 with CMake](https://doc.qt.io/qt-6/cmake-qt5-and-qt6-compatibility.html):
 
-Then we can write our `CMakeLists.txt` file:
+```cmake,ignore
+{{#include ../../../examples/qml_minimal/CMakeLists.txt:book_cmake_setup}}
+```
 
+From the Qt CMake target, find the location of the qmake executable. cxx-qt-build will need this later.
 ```cmake,ignore
-{{#include ../../../examples/qml_minimal/CMakeLists.txt:book_tutorial_cmake_full}}
+{{#include ../../../examples/qml_minimal/CMakeLists.txt:book_cmake_find_qmake}}
 ```
 
-This looks like a lot, but it is actually just a fairly standard CMake file for building a Qt application.
+Locate [Corrosion](https://github.com/corrosion-rs/corrosion), a tool for integrating Rust libraries into CMake.
+If Corrosion is not installed, automatically download it:
 
-The difference here are these lines:
 ```cmake,ignore
-{{#include ../../../examples/qml_minimal/CMakeLists.txt:book_tutorial_cmake_diff_1}}
-{{#include ../../../examples/qml_minimal/CMakeLists.txt:book_tutorial_cmake_diff_2}}
+{{#include ../../../examples/qml_minimal/CMakeLists.txt:book_cmake_find_corrosion}}
 ```
 
-Which will do the code generation and include it into the C++ build.
+Use Corrosion to create a CMake library target for the Rust library. cxx-qt requires a few more steps beyond using
+a typical Rust library with Corrosion:
+```cmake,ignore
+{{#include ../../../examples/qml_minimal/CMakeLists.txt:book_cmake_use_corrosion}}
+```
+
+Finally, create the CMake executable target and link it to the Rust library:
+
+```cmake,ignore
+{{#include ../../../examples/qml_minimal/CMakeLists.txt:book_cmake_executable}}
+```
 
-An important thing to note here is that CMake must be able to resolve the call to `include(CxxQt)`.
-For this to work, you'll want to clone the [CXX-Qt repository](https://github.com/KDAB/cxx-qt/) and add the `CxxQt.cmake` file to the `CMAKE_MODULE_PATH` CMake variable.
-An easy way to achieve this is by using CMake's `-D` option.
-For some alternatives, see the [CMake concepts chapter](../concepts/cmake.md).
+Build the project like any other CMake project:
 
-Therefore building our project can be done like this:
 ```shell
-$ mkdir build && cd build
-$ cmake -DCMAKE_MODULE_PATH="<path-to-cxx-qt-repo>/cmake" ..
-$ cmake --build .
+$ cmake -S . -B build
+$ cmake --build build
 ```
 If this fails for any reason, take a look at the [`examples/qml_minimal`](https://github.com/KDAB/cxx-qt/tree/main/examples/qml_minimal) folder, which contains the complete example code.
 
 This should now configure and compile our project.
 If this was successful, you can now run our little project.
 ```shell
-$ ./qml_minimal
+$ build/qml_minimal
 ```
 
 You should now see the two Labels that display the state of our `MyObject`, as well as the two buttons to call our two Rust functions.
```

Comment:
To the "Success" and "Further reading" part below, maybe add link to the `6-cargo-executable` example?

Reply:
I don't linking is necessary because it's clear from the table of contents and pressing next. But I could add a sentence or two to transition to the next chapter.

Reply:
Maybe just a statement somewhere saying, how 5 describing using CMake. But it is possible not just use Cargo and 6 describes this?